assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -50,6 +52,7 @@ import javax.lang.model.util.Types
 import javax.tools.Diagnostic.Kind.ERROR
 import javax.tools.Diagnostic.Kind.WARNING
 
+@IncrementalAnnotationProcessor(AGGREGATING)
```

Comment:
Why is this aggregating? We generate one type per annotated class.

Reply:
Ah, sorry, I was responding from my phone earlier. Had a chance to take a full look at it.
According to the [Gradle Incremental Annotation Processing documentation](https://docs.gradle.org/5.5/userguide/java_plugin.html#sec:incremental_annotation_processing), isolated annotation processors must meet the following condition:
> They must provide exactly one originating element for each file generated with the Filer API. If zero or many originating elements are provided, Gradle will recompile all source files.

In this case, we specify two originating elements on line 272 and 273 of this file.
https://github.com/square/AssistedInject/pull/100/files/57728eb7ff64f79a440bf47466b6125ba899291e#diff-6d2e2839b911a0a5c324c1944e5c7dd6R272

That is why I made this an aggregating annotation processor.

Reply:
The factory type can be removed from that since it's always enclosed in the target type.

Reply:
Updated! Nice catch